tests/api/ocsp.c

```diff
@@ -0,0 +1,539 @@
+    int reqSz, unsigned char** respBuf)
+{
+    struct ocsp_cb_ctx* cb_ctx = (struct ocsp_cb_ctx*)ctx;
+    (void)url;
+    (void)urlSz;
+    (void)req;
+    (void)reqSz;
+
+    *respBuf = cb_ctx->response;
+    return cb_ctx->responseSz;
+}
+
+static int test_ocsp_response_with_cm(struct test_conf* c)
+{
+    EXPECT_DECLS;
+    WOLFSSL_CERT_MANAGER* cm = NULL;
+    struct ocsp_cb_ctx cb_ctx;
+    int ret;
+
+    cm = wolfSSL_CertManagerNew();
+    ExpectPtrNE(cm, NULL);
+    ret = wolfSSL_CertManagerEnableOCSP(cm,
+        WOLFSSL_OCSP_URL_OVERRIDE | WOLFSSL_OCSP_NO_NONCE);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    ret = wolfSSL_CertManagerSetOCSPOverrideURL(cm, "http://foo.com");
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    cb_ctx.response = (byte*)c->resp;
+    cb_ctx.responseSz = c->respSz;
+    ret = wolfSSL_CertManagerSetOCSP_Cb(cm, ocsp_cb, NULL, (void*)&cb_ctx);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    /* add ca in cm */
+    if (c->ca0 != NULL) {
+        ret = wolfSSL_CertManagerLoadCABuffer(cm, c->ca0, c->ca0Sz,
+            WOLFSSL_FILETYPE_ASN1);
+        ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    }
+    if (c->ca1 != NULL) {
+        ret = wolfSSL_CertManagerLoadCABuffer(cm, c->ca1, c->ca1Sz,
+            WOLFSSL_FILETYPE_ASN1);
+        ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    }
+    /* check cert */
+    ret = wolfSSL_CertManagerCheckOCSP(cm, c->targetCert, c->targetCertSz);
+    wolfSSL_CertManagerFree(cm);
+    return ret;
+}
+
+int test_ocsp_response_parsing(void)
+{
+    struct test_conf conf;
+    int ret;
+    EXPECT_DECLS;
+    conf.resp = (unsigned char*)resp;
+    conf.respSz = sizeof(resp);
+    conf.ca0 = root_ca_cert_pem;
+    conf.ca0Sz = sizeof(root_ca_cert_pem);
+    conf.ca1 = NULL;
+    conf.ca1Sz = 0;
+    conf.targetCert = intermediate1_ca_cert_pem;
+    conf.targetCertSz = sizeof(intermediate1_ca_cert_pem);
+    ret = test_ocsp_response_with_cm(&conf);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+
+    conf.resp = (unsigned char*)resp_multi;
+    conf.respSz = sizeof(resp_multi);
+    conf.ca0 = root_ca_cert_pem;
+    conf.ca0Sz = sizeof(root_ca_cert_pem);
+    conf.ca1 = NULL;
+    conf.ca1Sz = 0;
+    conf.targetCert = intermediate1_ca_cert_pem;
+    conf.targetCertSz = sizeof(intermediate1_ca_cert_pem);
+    ret = test_ocsp_response_with_cm(&conf);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+
+    conf.resp = (unsigned char*)resp_bad_noauth;
+    conf.respSz = sizeof(resp_bad_noauth);
+    conf.ca0 = root_ca_cert_pem;
+    conf.ca0Sz = sizeof(root_ca_cert_pem);
+    conf.ca1 = ca_cert_pem;
+    conf.ca1Sz = sizeof(ca_cert_pem);
+    conf.targetCert = server_cert_pem;
+    conf.targetCertSz = sizeof(server_cert_pem);
+    ret = test_ocsp_response_with_cm(&conf);
+#ifndef WOLFSSL_NO_OCSP_ISSUER_CHECK
+    ExpectIntNE(ret, WOLFSSL_SUCCESS);
+#else
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+#endif
+    return EXPECT_SUCCESS();
+}
+#else  /* HAVE_OCSP */
+int test_ocsp_response_parsing(void) { return TEST_SKIPPED; }
```

Comment:
Format normally. Rest as well.
```suggestion
    int test_ocsp_response_parsing(void)
    {
        return TEST_SKIPPED;
    }
```

Reply:
`\n` before `{`

Reply:
`test_ocsp_status_callback` left.